packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -108,6 +108,52 @@ export type PreferencesState = {
   useTokenDetection: boolean;
 };
 
+/**
+ * Get the default PreferencesController state.
+ *
+ * @returns The default PreferencesController state.
+ */
+export function getDefaultPreferencesState() {
```

Comment:
What kind of mutations are we trying to avoid in this case? 

Would a [const assertion](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) cover those cases or do we expect the need for runtime protection against these mutations?

```typescript
const defaultState = { foo: 'bar' } as const;

class PreferencesController {
  constructor() {
    this.defaultState = { ...defaultState };
  }
}
```

Reply:
Runtime. I found that the feature flags were being mutated in tests, which `const` would not prevent.